File: Python/Zolw.py
from Zwierze import Zwierze
from Organizm import Organizm

class Zolw(Zwierze):
    def __init__(self, x, y, dz, swiat):
        super().__init__(x, y, dz, 2, 1, swiat.IDZOLW, swiat)

    def rysowanie(self):
        return "#216d33"

    def nazwa(self):
        return "Zolw"

    def stworz_nowe_zwierze(self, x, y, sw):
        org = Zolw(x, y, True, sw)

    def CzyRuszac(self):
        random = self.swiat.rand.randint(0, 3)

        if random == 3:
            return True
        return False

    def CzyOdbijaAtak(self, atakujacy):
        if atakujacy.GetSila() < 5:
            return 2
        return 0

File: C++/Zolw.cpp
#include <iostream>
#include <windows.h>
#include "Zolw.h"

Zolw::Zolw(int x, int y, Swiat* swiat) : Zwierze(x, y, 2, 1, IDZOLW, swiat)
{
}

Zolw::Zolw(int x, int y, bool dz, Swiat* swiat) : Zwierze(x, y, dz, 2, 1, IDZOLW, swiat)
{
}

void Zolw::rysowanie()
{
	HANDLE col;
	col = GetStdHandle(STD_OUTPUT_HANDLE);
	SetConsoleTextAttribute(col, KOLORZOLW);
	std::cout << "Z";
}

std::string Zolw::nazwa()
{
	return "Z";
}

void Zolw::stworz_nowe_zwierze(int x, int y, Swiat* sw)
{
	Organizm* org = new Zolw(x, y, true, sw);
}

bool Zolw::CzyRuszac()
{
	int random = rand() % 4;

	if (random == 3)
		return true;
	return false;
}

int Zolw::CzyOdbijaAtak(Organizm* atakujacy)
{
	if (atakujacy->GetSila() < 5)
		return 2;
	return 0;
}

Zolw::~Zolw()
{
}